spec/requests/api/graphql/graphql_controller_spec.rb

```diff
@@ -0,0 +1,110 @@
+RSpec.describe API::GraphqlController, issues: ['railgun#147'] do
+
+  context 'Administrator can use users and user_roles' do
+    let(:authenticated_header) {
+      { 'Authorization' => "Bearer #{create(:administrator).api_token}" }
+    }
+
+    describe 'post /api/graphql#execute' do
+      it 'valid request returns users' do
+        users = create_list(:user, 2)
+        post '/api/graphql', headers: authenticated_header, params: { query: '{ users { id email }}' }
+
+        expect(response.body).to match(users.first.email)
+      end
+
+      it 'valid request returns user' do
+        user = create(:user)
+        post '/api/graphql', headers: authenticated_header, params: { query: "{ user(id: \"#{user.id}\") { id email }}" }
+
+        expect(response.body).to match(user.email)
+      end
+
+      it 'valid request returns user_roles' do
+        user = create(:owner)
+        post '/api/graphql', headers: authenticated_header, params: { query: "{ user_role(user_id: #{user.id}) { roles }}" }
+
+        expect(response.body).to match('owner')
+      end
+    end
+  end
+
+  context 'Developer can use snippets' do
+    let(:authenticated_header) {
+      { 'Authorization' => "Bearer #{create(:developer).api_token}" }
+    }
+
+    describe 'post /api/graphql#execute' do
+      it 'valid request returns snippets' do
+        snippets = create_list(:snippet, 2)
+        post '/api/graphql', headers: authenticated_header, params: { query: '{ snippets { slug }}' }
+
+        expect(response.body).to match(snippets.first.slug)
+      end
+
+      it 'valid request returns snippets' do
+        snippet = create(:snippet)
+        post '/api/graphql', headers: authenticated_header, params: { query: "{ snippet(slug: \"#{snippet.slug}\") { slug text }}" }
+
+        expect(response.body).to match(snippet.text)
+      end
+
+      it 'Create Snippet' do
+        post '/api/graphql', headers: authenticated_header, params: { query: "mutation{addSnippet(snippet: {slug: \"qwe2\", text: \"My graphql snippet\"}) { slug text }}" }
+
+        expect(Snippet.find_by_slug('qwe2').text).to eq('My graphql snippet')
+      end
+    end
+  end
+
+  context 'Editor can use medias and pages' do
+    let(:authenticated_header) {
+      { 'Authorization' => "Bearer #{create(:editor).api_token}" }
+    }
+
+    describe 'post /api/graphql#execute' do
+      it 'valid request returns medias' do
+        medias = create_list(:media, 2)
+        post '/api/graphql', headers: authenticated_header, params: { query: '{ medias { slug file_file_name }}' }
+
+        expect(response.body).to match(medias.first.slug)
+      end
+
+      it 'valid request returns media' do
+        media = create(:media)
+        post '/api/graphql', headers: authenticated_header, params: { query: "{ media(slug: \"#{media.slug}\") { slug file_file_name }}" }
+
+        expect(response.body).to match(media.file_file_name)
+      end
+
+      # TODO adding a real file
+      it 'Create Media' do
```

Comment:
Something like this should work: http://mgiroux.me/2015/uploading-files-using-relay-with-rails/